src/main/java/bio/terra/service/journal/JournalService.java

```diff
@@ -0,0 +1,208 @@
+   * @return UUID of the entry created.
+   */
+  public UUID journalCreate(
+      @NotNull AuthenticatedUserRequest user,
+      @NotNull UUID resourceKey,
+      @NotNull IamResourceType resourceType,
+      String note,
+      Map changeMap) {
+    return journal(EntryType.CREATE, user, resourceKey, resourceType, note, changeMap, false);
+  }
+
+  /**
+   * Create a journal entry that indicates a resource was created and optionally deletes matching
+   * journal entries that may have existed. Billing profiles are an example of a domain object that
+   * supports UUID reuse.
+   *
+   * @param user The user performing the action being journaled
+   * @param resourceKey The UUID of the domain object the journal entry pertains to
+   * @param resourceType The {@code IamResourceType} that maps to the domain object the journal
+   *     entry pertains to
+   * @param note An optional message describing the event
+   * @param changeMap An optional map of changes to store with the event
+   * @param clearHistory A boolean to indicate that any prior journal entries should be removed on
+   *     this create.
+   * @return UUID of the entry created.
+   */
+  public UUID journalCreate(
+      @NotNull AuthenticatedUserRequest user,
+      @NotNull UUID resourceKey,
+      @NotNull IamResourceType resourceType,
+      String note,
+      Map changeMap,
+      boolean clearHistory) {
+    return journal(
+        EntryType.CREATE, user, resourceKey, resourceType, note, changeMap, clearHistory);
+  }
+
+  /**
+   * Create a journal entry that indicates a resource was updated or had an action performed that
+   * was worth noting.
+   *
+   * @param user The user performing the action being journaled
+   * @param resourceKey The UUID of the domain object the journal entry pertains to
+   * @param resourceType The {@code IamResourceType} that maps to the domain object the journal
+   *     entry pertains to
+   * @param note An optional message describing the event
+   * @param changeMap An optional map of changes to store with the event
+   * @return UUID of the entry created.
+   */
+  public UUID journalUpdate(
+      @NotNull AuthenticatedUserRequest user,
+      @NotNull UUID resourceKey,
+      @NotNull IamResourceType resourceType,
+      String note,
+      Map changeMap) {
+    return journal(EntryType.UPDATE, user, resourceKey, resourceType, note, changeMap, false);
+  }
+
+  /**
+   * Create a journal entry that indicates a resource was deleted.
+   *
+   * @param user The user performing the action being journaled
+   * @param resourceKey The UUID of the domain object the journal entry pertains to
+   * @param resourceType The {@code IamResourceType} that maps to the domain object the journal
+   *     entry pertains to
+   * @param note An optional message describing the event
+   * @param changeMap An optional map of changes to store with the event
+   * @return UUID of the entry created.
+   */
+  public UUID journalDelete(
```

Comment:
Naming things is hard, but I find it confusing that `journalDelete()` _creates_ a `delete` journal entry. Would it be too much to have a `createEntry()` API that required the caller to pass in the type of journal entry they're creating?

(Also I think it's redundant to use the class name in a class method, since every use of the method will be via a variable or field that already has `journal` in its name.)

Reply:
Agreed that naming things is hard.  In an attempt to make things clearer, I discussed this with our team today.  We settled on "record" instead of journal.